app/forms/AssistanceDetailsForm.scala

```diff
@@ -60,7 +43,66 @@ object AssistanceDetailsForm {
                    needsSupportForOnlineAssessment: String,
                    needsSupportForOnlineAssessmentDescription: Option[String],
                    needsSupportAtVenue: String,
-                   needsSupportAtVenueDescription: Option[String]
-  )
+                   needsSupportAtVenueDescription: Option[String]) {
+
+    def exchange: AssistanceDetails = {
+      AssistanceDetails(
+        hasDisability,
+        hasDisabilityDescription,
+        guaranteedInterview.map {
+          case "Yes" => true
+          case "No" => false
+          case _ => false
+        },
+        needsSupportForOnlineAssessment match {
+          case "Yes" => true
+          case "No" => false
+          case _ => false
+        },
+        needsSupportForOnlineAssessmentDescription,
+        needsSupportAtVenue match {
+          case "Yes" => true
+          case "No" => false
+          case _ => false
+        },
+        needsSupportAtVenueDescription
+      )
+    }
 
+    def sanitizeData: AssistanceDetailsForm.Data = {
+      AssistanceDetailsForm.Data(
+        hasDisability,
+        if (hasDisability == "Yes") hasDisabilityDescription else None,
+        if (hasDisability == "Yes") guaranteedInterview else None,
+        needsSupportForOnlineAssessment,
+        if (needsSupportForOnlineAssessment == "Yes") needsSupportForOnlineAssessmentDescription else None,
+        needsSupportAtVenue,
+        if (needsSupportAtVenue == "Yes") needsSupportAtVenueDescription else None)
+    }
+  }
+
+  object Data{
+    def apply(ad: AssistanceDetails): Data = {
+      AssistanceDetailsForm.Data(
+        ad.hasDisability,
+        ad.hasDisabilityDescription,
+        ad.guaranteedInterview.map {
```

Comment:
Just observed at few places we are converting from "Yes" to "true" and vice versa using pattern match.
Instead if we stick to actual "true" and "false" strings then we can use this Mappings.nonemptyBooleanText formatter and apply simple toBoolean conversion.